src/wifi/model/wifi-mac.cc

```diff
@@ -478,49 +478,151 @@ WifiMac::Configure80211ad (void)
 }
 
 void
-WifiMac::ConfigureDcf (Ptr<DcaTxop> dcf, uint32_t cwmin, uint32_t cwmax, bool isDsss, AcIndex ac)
+WifiMac::ConfigureDcf (Ptr<DcaTxop> dcf, uint32_t cwmin, uint32_t cwmax, bool isDsss, AcIndex ac, bool isDmgSupported)
 {
   NS_LOG_FUNCTION (this << dcf << cwmin << cwmax << isDsss << ac);
-  /* see IEEE802.11 section 7.3.2.29 */
+  /* see IEEE802.11-2016 Table 9-137 */
   switch (ac)
     {
     case AC_VO:
       dcf->SetMinCw ((cwmin + 1) / 4 - 1);
       dcf->SetMaxCw ((cwmin + 1) / 2 - 1);
       dcf->SetAifsn (2);
-      if (isDsss)
+      switch (m_standard)
         {
-          dcf->SetTxopLimit (MicroSeconds (3264));
-        }
-      else
-        {
-          dcf->SetTxopLimit (MicroSeconds (1504));
+        case WIFI_PHY_STANDARD_80211ad:
+          dcf->SetTxopLimit (MicroSeconds (0));
+          break;
+        case WIFI_PHY_STANDARD_80211ax_5GHZ:
+        case WIFI_PHY_STANDARD_80211ac:
+        case WIFI_PHY_STANDARD_80211n_5GHZ:
+        case WIFI_PHY_STANDARD_80211ax_2_4GHZ:
+        case WIFI_PHY_STANDARD_80211n_2_4GHZ:
+        case WIFI_PHY_STANDARD_80211g:
+        case WIFI_PHY_STANDARD_holland:
+        case WIFI_PHY_STANDARD_80211a:
+          dcf->SetTxopLimit (MicroSeconds (2080));
+          break;
+        case WIFI_PHY_STANDARD_80211_10MHZ:
+        case WIFI_PHY_STANDARD_80211_5MHZ:
+        case WIFI_PHY_STANDARD_80211b:
+          if (isDsss)
+            {
+              dcf->SetTxopLimit (MicroSeconds (3264));
+            }
+          else
+            {
+              dcf->SetTxopLimit (MicroSeconds (2080));
+            }
+          break;
+        default:
+          NS_FATAL_ERROR ("Unsupported WifiPhyStandard in WifiMac::ConfigureDcf ()");
```

Comment:
to make it easier to compare it to the standard, please add comments explaining which clauses they refer to